blog/models.py

```diff
@@ -127,6 +128,54 @@ class BlogArticlePage(MixinSeoFields, Page, MixinPageMethods, GoogleAdsMixin):
         StreamFieldPanel("body"),
     ]
 
+    @cached_property
+    def intro(self) -> str:
+        paragraph_text = self._get_paragraph_text_for_intro(MAX_BLOG_ARTICLE_INTRO_LENGTH)
+        intro_text = ""
+        if len(paragraph_text) > 0:
+            words = paragraph_text.split()
+            word_index = 0
+            new_text = words[word_index]
+            while len(new_text) < MAX_BLOG_ARTICLE_INTRO_LENGTH:
+                intro_text = new_text
+                word_index += 1
+                new_text = f"{intro_text} {words[word_index]}"
+        return intro_text
+
+    def _get_paragraph_text_for_intro(self, character_limit: int) -> str:
+        body_stream_child_index = self._get_index_of_next_nearest_paragraph()
+        paragraph_text = (
+            ""
+            if body_stream_child_index is None
+            else self._strip_rich_text_of_html_tags(
+                self.body[body_stream_child_index].value.source  # pylint:disable=unsubscriptable-object
+            )
+        )
+        while body_stream_child_index is not None and len(paragraph_text) < character_limit:
+            body_stream_child_index = self._get_index_of_next_nearest_paragraph(body_stream_child_index + 1)
+            if body_stream_child_index is not None:
+                next_paragraph_child_text = self.body[  # pylint:disable=unsubscriptable-object
+                    body_stream_child_index
+                ].value.source
+                paragraph_text += f" {self._strip_rich_text_of_html_tags(next_paragraph_child_text)}"
+        return paragraph_text
+
+    def _get_index_of_next_nearest_paragraph(self, body_stream_child_index: int = 0) -> Optional[int]:
+        while (
+            body_stream_child_index < len(self.body)
+            and self.body[body_stream_child_index].block.name != "paragraph"  # pylint:disable=unsubscriptable-object
+        ):
+            body_stream_child_index += 1
+        if body_stream_child_index >= len(self.body):
+            return None
+        return body_stream_child_index
+
+    @staticmethod
+    def _strip_rich_text_of_html_tags(text: str) -> str:
```

Comment:
Did you try to use `strip_tags` from `django.utils.html`? Maybe it is working the same? Can you look at it?

Reply:
Sure, I'll check it out 👍

Reply:
Done.